### DISCUSSION TITLE
Empty html report from IfcTester in Python script

### DISCUSSION THREAD

**Discussion by Zeraxel** [IfcOpenShell/IfcOpenShell#3856] [Q&A] [answered]
Hi everyone! I have some trouble with getting html report, using ifcTester in python script

Thats my code
```

import  ifcopenshell
import ifctester
from ifctester import reporter, ids

my_ifc = ifcopenshell.open("D:\ifcOpenShellTest\IfcTester\IDS_TestFile_001.ifc")
my_ids = ids.open("D:\ifcOpenShellTest\IfcTester\IDS_TestIds_001.ids", validate=True)

my_ids.validate(my_ifc)

ifctester.reporter.Html(my_ids).to_file("D:\ifcOpenShellTest\IfcTester\jest.html")
```

But when i open html file, it's empty. What i doing wrong? Please help.

When i using Console, it works right. But in HTML, JSON and txt i can't see results.

**Comment by Moult** [comment]
Can you try with forward slashes instead of backslashes?

**Reply by Zeraxel** [reply]
Yes, it make the same result...
screenshot link: https://imgur.com/a/146yFTT

**Reply by Moult** [reply, marked as answer]
You need to call report() first prior to to_file().

```
reporter = ifctester.reporter.Html(my_ids)
reporter.report()
reporter.to_file('your/path/here.html')
```

**Reply by Zeraxel** [reply]
Great, that's work. Thanks!
